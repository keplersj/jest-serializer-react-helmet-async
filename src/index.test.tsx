import * as React from "react";
import renderer from "react-test-renderer";
import { Helmet, HelmetProvider } from "react-helmet-async";
import serializer from ".";

const TestHead = ({
  children,
}: React.PropsWithChildren<{}>): React.ReactElement => (
  <>
    <Helmet>
      <html lang="en" />
      <title>Hello, Helmet!</title>
      <meta name="description" content="This is a unit test!" />
    </Helmet>
    {children}
  </>
);

describe("Serialization function", () => {
  it("serializes React Helmet Async context correctly", () => {
    expect.addSnapshotSerializer(serializer);
    const context: { helmet?: object } = {};
    const tree = renderer
      .create(
        <HelmetProvider context={context}>
          <TestHead>
            <div>
              <span>Hello, Jest!</span>
            </div>
          </TestHead>
        </HelmetProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(context.helmet).toMatchSnapshot();
  });
});

describe("Serialization module", () => {
  it("matches expectations", () => {
    expect(serializer).toMatchSnapshot();
  });
});
