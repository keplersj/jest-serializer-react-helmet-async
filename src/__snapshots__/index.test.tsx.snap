// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serialization function serializes React Helmet Async context correctly 1`] = `
<div>
  <span>
    Hello, Jest!
  </span>
</div>
`;

exports[`Serialization function serializes React Helmet Async context correctly 2`] = `
<html
  lang="en"
>
  <head>
    <meta
      content="This is a unit test!"
      data-rh={true}
      name="description"
    />
    <title
      data-rh={true}
    >
      Hello, Helmet!
    </title>
  </head>
  <body />
</html>
`;

exports[`Serialization module matches expectations 1`] = `
Object {
  "print": [Function],
  "test": [Function],
}
`;
